src/Components/Server/src/ComponentHub.cs

```diff
@@ -99,16 +111,28 @@ public async Task<bool> ConnectCircuit(string circuitId)
             {
                 CircuitHost = circuitHost;
 
+                circuitHost.Initialized();
+                InitializeServicesAfterPrerender(circuitHost);
+
                 // Dispatch any buffered renders we accumulated during a disconnect.
                 // Note that while the rendering is async, we cannot await it here. The Task returned by ProcessBufferedRenderBatches relies on
                 // OnRenderCompleted to be invoked to complete, and SignalR does not allow concurrent hub method invocations.
-                _ = circuitHost.Renderer.ProcessBufferedRenderBatches();
+                var _ = CircuitHost.Renderer.InvokeAsync(() => circuitHost.Renderer.ProcessBufferedRenderBatches());
```

Comment:
I think this makes more sense as a method on the circuit host. Right now we're accumulating a lot of complexity in two places, we should try to contain it in one.

Reply:
done